src/app/static/src/tests/components/plots/mapControl.test.ts

```diff
@@ -33,42 +42,44 @@ describe("Map control component", () => {
     it("renders tree selects with expected properties", () => {
         const wrapper = mount(MapControl, {props, global: {plugins: [store]}});
 
-        expect(wrapper.findAllComponents(TreeSelect)[0].props("searchable")).toBe(false);
-        expect(wrapper.findAllComponents(TreeSelect)[0].props("multiple")).toBe(false);
-        expect(wrapper.findAllComponents(TreeSelect)[0].props("clearable")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[0].props("searchable")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[0].props("multiple")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[0].props("clearable")).toBe(false);
 
-        expect(wrapper.findAllComponents(TreeSelect)[1].props("searchable")).toBe(false);
-        expect(wrapper.findAllComponents(TreeSelect)[1].props("multiple")).toBe(false);
-        expect(wrapper.findAllComponents(TreeSelect)[1].props("clearable")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[1].props("searchable")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[1].props("multiple")).toBe(false);
+        expect(wrapper.findAllComponents(Treeselect)[1].props("clearable")).toBe(false);
     });
 
     it("renders indicator options", () => {
         const wrapper = mount(MapControl, {props, global: {plugins: [store]}});
 
-        expect(wrapper.findAllComponents(TreeSelect)[0].props("options"))
+        expect(wrapper.findAllComponents(Treeselect)[0].props("options"))
             .toStrictEqual([{id: "art_coverage", label: "ART coverage"},
                 {id: "prevalence", label: "Prevalence"}]);
     });
 
     it("renders detail options", () => {
         const wrapper = mount(MapControl, {props, global: {plugins: [store]}});
 
-        expect(wrapper.findAllComponents(TreeSelect)[1].props("options"))
+        expect(wrapper.findAllComponents(Treeselect)[1].props("options"))
             .toStrictEqual([{id: 4, label: "Admin Level 4"},
                 {id: 5, label: "Admin Level 5"}]);
     });
 
 
-    it("emits indicator-changed event with indicator", () => {
+    it("emits indicator-changed event with indicator", async () => {
         const wrapper = mount(MapControl, {props, global: {plugins: [store]}});
-        wrapper.findAllComponents(TreeSelect)[0].vm.$emit("input", "art_coverage");
-        expect(wrapper.emitted("indicator-changed")![0][0]).toBe("art_coverage");
+        wrapper.findAllComponents(HintTreeSelect)[0].vm.$emit("update:model-value", "art_coverage");
+        await nextTick();
```

Comment:
Are you sure you need this `nextTick` - I thought `emit`s were synchronous? I tried running locally without it and it passed.

Reply:
youre right! done